internal/querynode/segment.go

```diff
@@ -411,7 +411,8 @@ func (s *Segment) retrieve(plan *RetrievePlan) (*segcorepb.RetrieveResults, erro
 		return nil, err
 	}
 
-	sort.Sort(&byPK{result})
+	timeStamp := GetTimeStampFieldsData(result)
+	sort.Sort(&byTS{result, timeStamp})
```

Comment:
Please don't change the result order unless you change all the reducing logic along with it. Or the result after this PR is meaningless.